RealmSwift/Results.swift

```diff
@@ -422,6 +422,229 @@ extension Results: AssistedObjectiveCBridgeable {
     }
 }
 
+#if swift(>=4)
+// MARK: Typesafe Queries
+public protocol RealmPropertyType {
+    var _object: AnyObject { get }
+}
+
+public protocol RealmEquatablePropertyType: RealmPropertyType {}
+public protocol RealmComparablePropertyType: RealmEquatablePropertyType {}
+
+extension Results {
+    public func filter<P: Predicate>(_ expression: @autoclosure () -> P) -> Results<T> where P.ObjectType == T {
+        return filter(expression().predicate)
+    }
+
+    public func sorted<RealmObject: Object, RealmProperty: RealmPropertyType>(byKeyPath keyPath: KeyPath<RealmObject, RealmProperty>, ascending: Bool = true) -> Results<T> {
+        return sorted(by: [SortDescriptor(keyPath: keyPath._kvcKeyPathString!, ascending: ascending)])
```

Comment:
Is `_kvcKeyPathString` a private Swift standard library API? We may not be able to ship this PR unless it becomes part of the public API.

Reply:
Yes, `_kvcKeyPathString`  is undocumented yet. https://github.com/apple/swift/blob/master/stdlib/public/core/KeyPath.swift#L99-L105

Apple engineers realized the needs. I asked it at the WWDC lab. I made a feature request https://bugs.swift.org/browse/SR-5220. 

Probably would it be worth to provide as a separate microframework rather than the Realm API? Like RxRealm https://github.com/RxSwiftCommunity/RxRealm

Reply:
Yes, I think keeping this separate until the limitations and use of private API are addressed is the right approach.